packages/test/test-end-to-end-tests/src/test/detachedContainerTests.spec.ts

```diff
@@ -898,4 +898,51 @@ describeNoCompat("Detached Container", (getTestObjectProvider) => {
 			"Stored handle should match!!",
 		);
 	});
+
+	it("Compressors sync after detached container attaches and sends an op", async () => {
+		provider.reset();
+		const testConfig = {
+			...testContainerConfig,
+			runtimeOptions: {
+				...testContainerConfig.runtimeOptions,
+				enableRuntimeIdCompressor: true,
+			},
+		};
+		loader = provider.makeTestLoader(testConfig) as Loader;
+		const container = await loader.createDetachedContainer(provider.defaultCodeDetails);
+
+		// Get the root dataStore from the detached container.
+		const response = await container.request({ url: "/" });
+		const dataStore = response.value as ITestFluidObject;
+		const testChannel1 = await dataStore.getSharedObject<SharedCell>(sharedCellId);
+
+		// Generate an Id before attaching the container
+		(testChannel1 as any).runtime.idCompressor.generateCompressedId();
+		// Attach the container. The generated Id won't be synced until another op
+		// is sent after attaching becuase most DDSs don't send ops until they are attached.
+		await container.attach(request);
+
+		// Create another container to test sync
+		const url: any = await container.getAbsoluteUrl("");
+		const loader2 = provider.makeTestLoader(testConfig) as Loader;
+		const container2 = await loader2.resolve({ url });
+		const response2 = await container2.request({ url: "/" });
+		const dataStore2 = response2.value as ITestFluidObject;
+		const testChannel2 = await dataStore2.getSharedObject<SharedCell>(sharedCellId);
+		// Generate an Id in the second attached container and send an op to send the Ids
+		(testChannel2 as any).runtime.idCompressor.generateCompressedId();
+		testChannel2.set("value");
+
+		await provider.ensureSynchronized();
+
+		// Send an op in the first container to get its Ids sent
+		testChannel1.set("value2");
```

Comment:
What happens to the id generated in detached container if this op is not sent? Why is sending this necessary?

Reply:
Because we're piggybacking off of ops being sent by containers, an op has to be sent by the detached container after attaching to get the compressors to synchronize. The id is still present in the local compressor, it just won't be finalized until the container sends an op. That means you don't get a cluster allocated yet and would still have local Ids if you generated more ops.

Reply:
I see. Let me ask you another question - If a detached container never sent any ops after attach, will the ids generated in detached mode be consistent and work across clients? What about ids generated by other clients? Will this container still be able to work with them?
Note: Maybe you response answered these questions but it wasn't clear to me.

Reply:
The other clients wouldn't know about the ids generated in the detached container if it never sent an op after attaching. However, ids generated by other clients would reach the newly attached container as they'll get delivered like any other ops.  

Theoretically, the detached container should be able to finalize its IdRanges without sending an op because there aren't any other clients to worry about, but I haven't done that optimization as part of this PR and I'm not exactly sure how the implementation will look.